code/modules/food_and_drinks/pizzabox.dm

```diff
@@ -286,3 +287,43 @@
 	. = ..()
 	pizza = new /obj/item/reagent_containers/food/snacks/pizza/meat(src)
 	boxtag = "Meatlover's Supreme"
+
+//An anomalous pizza box that, when opened, produces the opener's favorite kind of pizza.
+/obj/item/pizzabox/infinite
+	resistance_flags = FIRE_PROOF | LAVA_PROOF | ACID_PROOF //hard to destroy
+	can_open_on_fall = FALSE
+	var/list/pizza_types = list(
+		/obj/item/reagent_containers/food/snacks/pizza/meat = 1,
+		/obj/item/reagent_containers/food/snacks/pizza/mushroom = 1,
+		/obj/item/reagent_containers/food/snacks/pizza/margherita = 1,
+		/obj/item/reagent_containers/food/snacks/pizza/sassysage = 0.8,
+		/obj/item/reagent_containers/food/snacks/pizza/vegetable = 0.8,
+		/obj/item/reagent_containers/food/snacks/pizza/donkpocket = 0.3,
+		/obj/item/reagent_containers/food/snacks/pizza/dank = 0.1) //pizzas here are weighted by chance to be someone's favorite
+	var/static/list/pizza_preferences
+
+/obj/item/pizzabox/infinite/Initialize()
+	. = ..()
+	if(!pizza_preferences)
+		pizza_preferences = list()
+
+/obj/item/pizzabox/infinite/examine(mob/user)
+	..()
+	if(isobserver(user))
+		to_chat(user, "<span class='deadsay'>This pizza box is anomalous, and will produce infinite pizza.</span>")
+
+/obj/item/pizzabox/infinite/attack_self(mob/living/user)
+	QDEL_NULL(pizza)
+	if(ishuman(user))
+		attune_pizza(user)
+	. = ..()
+
+/obj/item/pizzabox/infinite/proc/attune_pizza(mob/living/carbon/human/noms) //tonight on "proc names I never thought I'd type"
+	if(!pizza_preferences[noms.ckey])
+		pizza_preferences[noms.ckey] = pickweight(pizza_types)
+		if(noms.mind && noms.mind.assigned_role == "Botanist")

```

Comment:
I was about to ask that you add something so botanists always get weed pizza, but I see you already did